### PR TITLE
[QNN EP] QNN SDK 2.28.2

### PR DESCRIPTION
**Pull request by adrianlizarraga**
### Description
- Updates pipelines to use QNN SDK 2.28.2.241116.
- Re-enable LayerNormalization unit tests that failed with accuracy errors with the previous QNN SDK (2.28.0).
- Update QNN EP to no longer provide a dummy bias for LayerNorm if the QNN SDK version is >= 2.28.0.

### Motivation and Context
Use the latest QNN SDK. This version improves inference latency for certain customer models.

### CODE REVIEW

File: onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc [hunk 1]
```diff
@@ -87,10 +87,10 @@ Status LayerNormOpBuilder::ProcessInputs(QnnModelWrapper& qnn_model_wrapper,
     ORT_RETURN_IF_ERROR(ProcessInput(qnn_model_wrapper, inputs[BIAS_IDX], logger, input_names));
   }
 
-#if QNN_API_VERSION_MAJOR == 2 && (QNN_API_VERSION_MINOR >= 17)
+#if QNN_API_VERSION_MAJOR == 2 && QNN_API_VERSION_MINOR >= 17 && QNN_API_VERSION_MINOR <= 20
   if (!has_bias_input && IsNpuBackend(qnn_model_wrapper.GetQnnBackendType())) {
-    // Bias is implicit. QNN SDK 2.24+ (QNN API version 2.17+) has a validation bug for implicit bias inputs,
-    // so provide an explicit bias of all 0 (quantized int32).
+    // Bias is implicit. QNN SDK 2.24 to 2.27 (QNN API version 2.17 to 2.20) has a validation bug for
+    // implicit bias inputs, so provide an explicit bias of all 0 (quantized int32).
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/opbuilder/layer_norm_op_builder.cc:93]: We can't use preprocessor values to differentiate between 2.20.0 and 2.20.2, but I checked that this validation bug was fixed on 2.20.0 as well.

File: tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml [hunk 2]
```diff
@@ -2,12 +2,12 @@ parameters:
...
   displayName: QNN SDK Version
   type: string
-  default: 2.28.0.241029
+  default: 2.28.2.241116
 
 - name: build_config
   displayName: Build Configuration
   type: string
-  default: 'RelWithDebInfo'
+  default: 'Release'
```
Comment (adrianlizarraga) [inline tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml:10]: NOTE: Updated default build config to `Release` for ORT+QNN Nuget packages.

Comment (adrianlizarraga) [reply tools/ci_build/github/azure-pipelines/qnn-ep-nuget-packaging-pipeline.yml]: Reverted.

File: onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc [hunk 3]
```diff
@@ -1040,7 +1041,14 @@ Status QnnBackendManager::ExtractBackendProfilingInfo() {
   const QnnProfile_EventId_t* profile_events{nullptr};
   uint32_t num_events{0};
   Qnn_ErrorHandle_t result = qnn_interface_.profileGetEvents(profile_backend_handle_, &profile_events, &num_events);
-  ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result, "Failed to get profile events. Error: ", QnnErrorHandleToString(result));
+  if (!qnn_saver_path_.empty()) {  // Using QNN Saver backend
+    // QNN SDK 2.28.2 returns QNN_SAVER_ERROR_DUMMY_RETVALUE, but previous QNN versions return QNN_PROFILE_NO_ERROR.
+    // We accept both values.
+    ORT_RETURN_IF(QNN_PROFILE_NO_ERROR != result && QNN_SAVER_ERROR_DUMMY_RETVALUE != result,
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/qnn/builder/qnn_backend_manager.cc:1047]: NOTE: The `QNN_SAVER_ERROR_DUMMY_RETVALUE` enum has been available for QNN sdk versions as far back as 2.23 (maybe older). The QNN header states that this error code is returned from APIs where Saver has no capability to fulfill the request but can still record the API call in saver_output.c. Apparently, this value has always been returned from APIs like `QnnProfile_getEventData()`, but now it is also returned from `profileGetEvents()`.

**Review by HectorSVC — APPROVED** [review]

**Review by jywu-msft — APPROVED** [review]

**Comment by snnn** [comment]
This PR has been cherry-picked into the `rel-1.20.1` branch in PR #22845. Removing the `release:1.20.1` label.

**Pull request merged**